.github/workflows/build.yml

```diff
@@ -0,0 +1,33 @@
+# Automatically build the project and run any configured tests for every push
+# and submitted pull request. This can help catch issues that only occur on
+# certain platforms or Java versions, and provides a first line of defence
+# against bad commits.
+
+name: build
+on: [pull_request, push]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        # Use these Java versions
+        java: [
+          1.8,  # Minimum supported by Minecraft
+          11,   # Current Java LTS
+          15    # Latest version
+        ]
+        # and run on both Linux and Windows
+        os: [ubuntu-20.04, windows-latest]
+    runs-on: ${{ matrix.os }}
```

Comment:
I would just do ubuntu only, I dont think building on windows provides any benefit. Building on windows would double the amount of stuff to download, and increase the amount github has to build (I know its free, but dont want to wase it for the sake of it)

Reply:
I've run into platform-specific issues before, in my own projects and what I've seen in some other repos:

- checkstyle doesn't normalize path separators in its suppressions file, so that can end up only working on one OS
- some parts of path handling differ, between different line separators and handling of symlinks -- though these would only really be caught by unit tests
- When developing on a case-insensitive filesystem (like windows has), renaming a class or folder in a way that only changes its case will not be captured by `git`.

These can be caught by running CI builds on multiple operating systems.